src/webots/nodes/utils/WbObjectDetection.cpp

```diff
@@ -142,18 +142,18 @@ bool WbObjectDetection::isWithinBounds(const WbAffinePlane *frustumPlanes, const
   const WbBaseNode *referenceObject = boundingObject;
   if (useBoundingSphere)
     referenceObject = rootObject;
-  const WbTransform *transform = dynamic_cast<const WbTransform *>(referenceObject);
-  if (!transform)
-    transform = referenceObject->upperTransform();
-  assert(transform);
-  WbMatrix3 objectRotation = transform->rotationMatrix();
-  WbVector3 objectPosition = transform->position();
+  const WbPose *pose = dynamic_cast<const WbPose *>(referenceObject);
+  if (!pose)
+    pose = referenceObject->upperPose();
+  assert(pose);
+  WbMatrix3 objectRotation = pose->rotationMatrix();
+  WbVector3 objectPosition = pose->position();
 
   if (nodeType == WB_NODE_SHAPE) {
     const WbShape *shape = static_cast<const WbShape *>(boundingObject);
     boundingObject = shape->geometry();
     return isWithinBounds(frustumPlanes, boundingObject, objectSize, objectRelativePosition);
-  } else if (nodeType == WB_NODE_GROUP || nodeType == WB_NODE_TRANSFORM) {
+  } else if (nodeType == WB_NODE_GROUP || nodeType == WB_NODE_TRANSFORM || nodeType == WB_NODE_POSE) {
```

Comment:
The test `nodeType == WB_NODE_TRANSFORM` can be removed if Transform nodes can no longer be inserted in the boundingObject.